### ISSUE TITLE
`extraErrors` is overriding any current validation errors

### ISSUE THREAD

**Issue by juliankigwana** [rjsf-team/react-jsonschema-form#1595]
### Prerequisites

- [X] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [X] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

I have found a bug with the `extraErrors` implementation for displaying Async Errors  https://github.com/rjsf-team/react-jsonschema-form/blob/ae02f4f901cd0dd217ac5ddb213d99ec5cc41c0c/packages/core/docs/advanced-customization.md#async-errors
Any current validation errors (current or future) are not displayed as the extraErrors property overwrites them.

### Steps to Reproduce

1. Add a schema with some client validation to a field.
2. Invalidate the field so that the validation error is visible
3. Add some extraErrors to the same field.

#### Expected behavior

Both the validation and async errors should be displayed

#### Actual behavior

Only the async errors are displayed

### Version

Effects both `v2.0.0-alpha.1` and `v2.0.0-alpha.2`

### Solution

I was able to fix this problem in `v2.0.0-alpha.1`,  however I am having problems running any of the scripts in `v2.0.0-alpha.2` so, I cant build, deploy, commit or create a PR. 

The problem lies [here](https://github.com/rjsf-team/react-jsonschema-form/blob/561dff16f0033f3c8fa996b87cab8bd3f7dc53d6/packages/core/src/components/Form.js#L75) within the `Form.js` where it merges the `errorSchema` with the `props.extraErrors`. If we look inside of the `mergeObjects` utility we see that it takes a third argument which tells the function to concatenate any arrays. Without this argument the array is replaced. Setting the argument to `true` does as it says, and concatenates the error arrays of both objects.

<img width="574" alt="Screenshot 2020-02-19 at 12 26 40" src="https://user-images.githubusercontent.com/1236178/74840679-4e3e0380-531f-11ea-9825-8e8bc8672d98.png">

**Comment by epicfaace** [comment]
Thanks for reporting the issue -- what issues are you facing running the scripts in v2.0.0-alpha.2?

**Comment by juliankigwana** [comment]
I think perhaps the rjsf docs have been out of date since `yarn` was replaced by `lerna` back in ef8b7fce19c5271f75a849fa951ad62595314b00 . The doc still refer to the old scripts which have all been replaced.

https://github.com/rjsf-team/react-jsonschema-form/blob/master/docs/src/index.md

I have not worked with `lerna` before and I don't really where to start. I tried starting with the official docs on their website, but I was soon overwhelmed by information overload. I tried, but I couldn't get any of the scripts to run successfully.

When I try to run any of the lerna scrips, I get various errors that include something like:
```
lerna ERR! npm run test stderr:
sh: cross-env: command not found
```

When I try to create a pull request I get the following message:

```
git --no-optional-locks -c color.branch=false -c color.diff=false -c color.status=false -c diff.mnemonicprefix=false -c core.quotepath=false -c credential.helper=sourcetree commit -q -F /var/folders/8s/npnkx0ts3_51wkg5km5kddtc0000gn/T/SourceTreeTemp.xenMD5 
Info: can't find node in PATH, trying to find a node binary on your system
husky > pre-commit (node v10.17.0)
lerna notice cli v3.20.2
lerna info Executing command in 2 packages: "npm run precommit"
@rjsf/playground: > @rjsf/playground@2.0.0-alpha.2.2 precommit /Users/juliankigwana/Development/react-jsonschema-form/packages/playground
@rjsf/playground: > lint-staged
@rjsf/playground: sh: lint-staged: command not found
@rjsf/playground: npm ERR! code ELIFECYCLE
@rjsf/playground: npm ERR! syscall spawn
@rjsf/playground: npm ERR! file sh
@rjsf/playground: npm ERR! errno ENOENT
@rjsf/playground: npm ERR! @rjsf/playground@2.0.0-alpha.2.2 precommit: `lint-staged`
@rjsf/playground: npm ERR! spawn ENOENT
@rjsf/playground: npm ERR! 
@rjsf/playground: npm ERR! Failed at the @rjsf/playground@2.0.0-alpha.2.2 precommit script.
@rjsf/playground: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
@rjsf/playground: npm WARN Local package.json exists, but node_modules missing, did you mean to install?
@rjsf/playground: npm ERR! A complete log of this run can be found in:
@rjsf/playground: npm ERR!     /Users/juliankigwana/.npm/_logs/2020-02-28T16_37_01_789Z-debug.log
lerna ERR! npm run precommit exited 1 in '@rjsf/playground'
husky > pre-commit hook failed (add --no-verify to bypass)
Completed with errors, see above
```

**Comment by olzraiti** [comment]
#1528 would solve this.

**Comment by juliankigwana** [comment]
@epicfaace - thank you for the updates to the docs. I am now able to start the playground and core packages. I still cant start either `npm test` or `npm test-coverage`. The docs are not clear as to whether to run these in the root or a package. Either way they fail with errors such as:

In the core package
```
> @rjsf/core@2.0.0-alpha.2 test /Users/juliankigwana/Development/react-jsonschema-form/packages/core
> cross-env NODE_ENV=test mocha --require @babel/register --require ./test/setup-jsdom.js test/**/*_test.js
```

In the root:
```
 react-jsonschema-form@2.0.0-alpha.1 test /Users/juliankigwana/Development/react-jsonschema-form
> lerna run test

lerna notice cli v3.18.4
lerna info Executing command in 2 packages: "npm run test"
lerna ERR! npm run test exited 1 in '@rjsf/core'
lerna ERR! npm run test stdout:

> @rjsf/core@2.0.0-alpha.2 test /Users/juliankigwana/Development/react-jsonschema-form/packages/core
> cross-env NODE_ENV=test mocha --require @babel/register --require ./test/setup-jsdom.js test/**/*_test.js


lerna ERR! npm run test stderr:
sh: cross-env: command not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! @rjsf/core@2.0.0-alpha.2 test: `cross-env NODE_ENV=test mocha --require @babel/register --require ./test/setup-jsdom.js test/**/*_test.js`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the @rjsf/core@2.0.0-alpha.2 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/juliankigwana/.npm/_logs/2020-03-20T15_56_44_788Z-debug.log

lerna ERR! npm run test exited 1 in '@rjsf/core'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
npm ERR! Test failed.  See above for more details.
```

And I cant commit, even locally without getting:
```
husky > pre-commit (node v10.17.0)
lerna notice cli v3.18.4
lerna info Executing command in 2 packages: "npm run precommit"
@rjsf/playground: > @rjsf/playground@2.0.0-alpha.2.2 precommit /Users/juliankigwana/Development/react-jsonschema-form/packages/playground
@rjsf/playground: > lint-staged
@rjsf/playground: No staged files match any of provided globs.
@rjsf/core: > @rjsf/core@2.0.0-alpha.2 precommit /Users/juliankigwana/Development/react-jsonschema-form/packages/core
@rjsf/core: > lint-staged
@rjsf/core: sh: lint-staged: command not found
@rjsf/core: npm ERR! code ELIFECYCLE
@rjsf/core: npm ERR! syscall spawn
@rjsf/core: npm ERR! file sh
@rjsf/core: npm ERR! errno ENOENT
@rjsf/core: npm ERR! @rjsf/core@2.0.0-alpha.2 precommit: `lint-staged`
@rjsf/core: npm ERR! spawn ENOENT
@rjsf/core: npm ERR!
@rjsf/core: npm ERR! Failed at the @rjsf/core@2.0.0-alpha.2 precommit script.
@rjsf/core: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
@rjsf/core: npm ERR! A complete log of this run can be found in:
@rjsf/core: npm ERR!     /Users/juliankigwana/.npm/_logs/2020-03-20T16_12_29_420Z-debug.log
```

**Comment by epicfaace** [comment]
@juliankigwana can you first run `lerna bootstrap` from the root directory?

**Comment by juliankigwana** [comment]
Yes, I am able to run lerna bootstrap from the root directory without any problems. 

```
➜  react-jsonschema-form git:(test) lerna bootstrap
info cli using local version of lerna
lerna notice cli v3.18.4
lerna info Bootstrapping 3 packages
lerna info Installing external dependencies
lerna info Symlinking packages and binaries
lerna success Bootstrapped 3 packages
```

**Comment by epicfaace** [comment]
@juliankigwana I believe this should now be fixed by #1528 -- I can no longer reproduce this error on the latest playground. Let me know if you still see it though.

**Comment by juliankigwana** [comment]
Thank you @epicfaace, it's gone. 
Im still having a few building issues, but I'll log those separately. Thanks again.
